pandas/core/generic.py

```diff
@@ -2930,25 +2931,61 @@ def blocks(self):
         """Internal property, property synonym for as_blocks()"""
         return self.as_blocks()
 
-    def astype(self, dtype, copy=True, raise_on_error=True, **kwargs):
+    def astype(self, dtype, copy=True, inplace=False, raise_on_error=True,
+               **kwargs):
         """
         Cast object to input numpy.dtype
-        Return a copy when copy = True (be really careful with this!)
 
         Parameters
         ----------
-        dtype : numpy.dtype or Python type
+        dtype : numpy.dtype or Python type (to cast entire DataFrame to the
+            same type). Alternatively, {col: dtype, ...}, where col is a column
+            label and dtype is a numpy.dtype or Python type (to cast one or
+            more of the DataFrame's columns to column-specific types).
+        copy : deprecated; use inplace instead
+        inplace : boolean, default False
+            Modify the NDFrame in place (do not create a new object)
```

Comment:
actually I don't think we can have an ``inplace`` flag here. you can't really do this except in some extreme circumstances (e.g. you *can* inplace ints to floats), but its not worth supporting at all and is quite convoluted.

The copy flag was just to prevent copies if we are already the same dtype. SO let's leave all of that alone.